src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/SurveyService.java

```diff
@@ -0,0 +1,27 @@
+package uk.gov.ons.ctp.integration.contactcentresvc.service.impl;
+
+import java.util.UUID;
+import ma.glasnost.orika.MapperFacade;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.ons.ctp.common.error.CTPException;
+import uk.gov.ons.ctp.integration.contactcentresvc.model.Survey;
+import uk.gov.ons.ctp.integration.contactcentresvc.repository.SurveyRepositoryClient;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.SurveyDTO;
+
+@Service
+public class SurveyService {
+
+  @Autowired private SurveyRepositoryClient surveyRepoClient;
+
+  @Autowired private MapperFacade mapper;
+
+  public SurveyDTO getSurvey(UUID surveyId) throws CTPException {
```

Comment:
Moved.

Reply:
While I don't mind if the @Transactional needs to be at the endpoint level, I guess that we should have a practice of pushing the annotation down the stack to where it is actually needed. If an earlier method in the stack ie the endpoint subsequently uses another service/method, also with transaction, and it requires each to be within the same transaction, it could then use transaction itself and lower level transactions join it.

So agreed - lets push this down the stack